### ISSUE TITLE
empty line in platforms.txt parses as an empty platform name

### ISSUE THREAD

**Issue by erg** [factor/factor#1544]
It's time to remove lines with only whitespace in them in at least platforms.txt, maybe authors.txt. Also, maybe having one toml file instead of several files would be better.

**Comment by bpollack** [comment]
Concluding files with a newline is common on Unix systems for a very
practical reason (the prompt won't be on a newline afterwards if
when you print to stdout via things like cat, otherwise). This is
the opposite of the Windows convention, because the console always
starts the prompt on a fresh line. If we want to say that Factor
follows Windows conventions, that's fine, but I wanted to point out
that just slapping on a call to harvest might be the better
practical option for now.


On Sat, Mar 5, 2016, at 11:55 AM, Doug Coleman wrote:
> It's time to remove lines with only whitespace in them in at least
> platforms.txt, maybe authors.txt if we don't. Also, maybe having one
> toml file instead of several files would be better.


> — Reply to this email directly or view it on GitHub[1].


>



Links:

  1. https://github.com/factor/factor/issues/1544

**Comment by jonenst** [comment]
I don't have a PC to check but files shouldn't contain empty newlines even in Unix. It's just that a text file is defined as a series of lines, and a line ends with \n. So the last line should end with its \n (and that does not mean that there is a following empty line).

**Comment by erg** [comment]
The error message for an empty platforms.txt line looks like this:

```
Loading resource:basis/xmode/code2html/responder/responder.factor
bad-platform
name ""
(U) Quotation: [ c-to-factor => ]
    Word: c-to-factor
(U) Quotation: [ [ (get-catchstack) push ] dip call => (get-catchstack) pop* ]
(O) Word: command-line-startup
(O) Word: do-all
```

I don't see a good reason not to just remove empty lines in this case.

**Comment by erg** [comment]
Fixed by 6c4960737ccffccaf8af76b75efdd984071362a9.
